doc/design/tensor_array.md

```diff
@@ -1,9 +1,50 @@
 # Design for TensorArray
+## Background
+Steps are one of the core concepts of RNN. In each time step of RNN, there should be several input segments, states, and output segments; all these components act like arrays, for example, call `states[step_id]` will get the state in `step_id`th time step.
```

Comment:
I'd suggest changing the first paragraph as:

> This design doc presents the necessity of a new C++ class `TensorArray`. In addition to the very simple C++ implementation
>
> ```c++
> typedef std::vector<LoDTensor*> TensorArray;
> ```
> we also need to expose it to PaddlePaddle's Python API, because users would want to use it with our very flexible operators WhileOp.  An example for your reference:

Reply:
I agree with @wangkuiyi . We should introduce the `TensorArray` and then start describing its use case in the RNN.